qds_sdk/commands.py

```diff
@@ -1347,6 +1347,44 @@ def parse(cls, args):
         v["command_type"] = "DbTapQueryCommand"
         return v
 
+class JupyterNotebookCommand(Command):
+    usage = "jupyternotebookcmd <submit|run> [options]"
+
+    optparser = GentleOptionParser(usage=usage)
+    optparser.add_option("--path", dest="path", help="Path including name of the Jupyter notebook to be run with extension.")
+    optparser.add_option("--cluster-label", dest="label", help="Label of the cluster on which the this command should be run. If this parameter is not specified then label = “default” is used.")
+    optparser.add_option("--arguments", dest="arguments", help="Valid JSON to be sent to the notebook. Specify the parameters in notebooks and pass the parameter value using the JSON format. key is the parameter’s name and value is the parameter’s value. Supported types in parameters are string, integer, float and boolean.")
+    optparser.add_option("--print-logs", action="store_true", dest="print_logs", default=False, help="Fetch logs and print them to stderr.")
+    optparser.add_option("--print-logs-live", action="store_true", dest="print_logs_live", default=False, help="Fetch logs and print them to stderr while command is running.")
+
+    @classmethod
+    def parse(cls, args):
+        """
+        Parse command line arguments to construct a dictionary of command
+        parameters that can be used to create a command
+
+        Args:
+            `args`: sequence of arguments
+
+        Returns:
+            Dictionary that can be used in create method
+
+        Raises:
+            ParseError: when the arguments are not correct
+        """
+        try:
+            (options, args) = cls.optparser.parse_args(args)
+            if options.path is None:
+                raise ParseError("Notebook Path must be specified", cls.optparser.format_help())
```

Comment:
if the path is local might want to check if the file exists? if we are not supporting the local path then it's fine.

Reply:
not a local path. The api does check if the file with given path exists, and if not it will throw an error.